lib/_http_client.js

```diff
@@ -68,7 +68,11 @@ function ClientRequest(options, cb) {
   self.socketPath = options.socketPath;
   self.timeout = options.timeout;
 
-  var method = self.method = (options.method || 'GET').toUpperCase();
+  const _method = options.method;
```

Comment:
Maybe we can just move the `var method` declaration up here and re-assign on line 75, that way we don't have a second variable for the same thing?